src/core/crypto.js

```diff
@@ -2013,7 +2014,15 @@ var CipherTransformFactory = (function CipherTransformFactoryClosure() {
     this.encryptionKey = encryptionKey;
 
     if (algorithm >= 4) {
-      this.cf = dict.get('CF');
+      var cf = dict.get('CF');
+      if (isDict(cf)) {
```

Comment:
It might help to open the PDF file from the issue with http://brendandahl.github.io/pdf.js.utils/browser/ and have that as a reference when reading my comment below :-)

Note that the `Encrypt` dict (when it's present) is a top-level item, contained in the `Trailer` dict, of a PDF file. The entries inside the `Encrypt` dict should not be encrypted, which thus explains the changes this patch makes in `src/core/obj.js`.
Now depending on the type of encryption used by the PDF file, the `Encrypt` dict may (as is the case here) have a `CF` entry (which is itself a dict). If you trace through the code, you'll see that the `dict` variable in the code you commented on above is referring to the `Encoding` dict.
Hence we're already *know* that we're inside of the `Encoding` dict here, and the `isDict` check above is only needed to prevent an error in the case where a particular encoding has no `CF` entry.

It's actually the `CF` dict that is the main source of the (current) problem in the code, since the lookup in https://github.com/mozilla/pdf.js/blob/master/src/core/crypto.js#L2044 is triggering an infinite loop. Hence why my patch tries to ensure that the `Encrypt` dict, and the `CF` dict within, isn't trying to needlessly decrypt the something that the [PDF spec says shouldn't be encrypted](http://www.adobe.com/content/dam/Adobe/en/devnet/acrobat/pdfs/PDF32000_2008.pdf#page=65) in the first place. 

Note that the reason this hasn't been an issue before is that this particular PDF file has indirect objects in the `Encrypt` dict, something the PDF spec seem to suggest should not actually happen.

A bit long-winded perhaps, but I hope it makes sense!